spatial/r3/vector.go

```diff
@@ -49,6 +53,29 @@ func (p Vec) Cross(q Vec) Vec {
 	}
 }
 
+// Rotate returns a new vector, rotated by alpha around the provided axis.
+func (p Vec) Rotate(alpha float64, axis Vec) Vec {
+	if alpha == 0 || (axis == Vec{}) {
```

Comment:
This just explodes into NaN if axis is zero. I think that's OK. Maybe leaving the zero test for alpha though.

Reply:
ok.